src/components/auth/LinkedInLogin.vue

```diff
@@ -0,0 +1,59 @@
+<script setup lang="ts">
+import { onMounted } from 'vue'
+
+import { useRoute } from 'vue-router'
+
+import useAuthStore from '@/stores/auth.store'
+
+import { linkedInScopes } from '@/config/components/auth/linkedInPermissions.config'
+
+const route = useRoute()
+
+const props = defineProps({
+  brandId: {
+    type: String,
+    required: true
+  }
+})
+
+const authStore = useAuthStore()
+
+const linkedInClientId = import.meta.env.VITE_LINKEDIN_CLIENT_ID
+const redirectUri =
+  'https://d811-2800-bf0-8143-f0-c5f4-6c27-3530-85de.ngrok-free.app'
+
+function loginWithLinkedIn() {
+  const state = Math.random().toString(36).substring(2)
+  const scope = linkedInScopes.join(' ')
+  const linkedInAuthUrl = `https://www.linkedin.com/oauth/v2/authorization?response_type=code&client_id=${linkedInClientId}&redirect_uri=${redirectUri}&state=${state}&scope=${scope}`
+
+  window.location.href = linkedInAuthUrl
+}
+
+function sendAccessToken(code: string) {
+  authStore.sendLinkedInTokenToBackend(code, props.brandId)
+}
+
+onMounted(() => {
+  const code = route.query.code as string
```

Comment:
and here apply a little validation to avoid the attack mentioned above


```suggestion
  const code = route.query.code as string
  const returnedState = reoute.queary.state as string
  const storedState = sessionStorage.getItem('oauthState')

  if(!code || !returnedState) {
    console.error('field missing')
    return 
  }

  if (returnedState !== storedState)  {
    console.error('invalid state, possible attack CSRG')
    return
  }

  sessionStorage.removeItem('oauthState')

  sendAccessToken(code)

```


this could be more secure with cookies with security atribbutes as "httponly, secur, samesite" but could be thought in a future security feat